xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -1834,10 +1835,13 @@ CacheInfo CVideoPlayer::GetCachingTimes()
   double cache_need = std::max(0.0, remain - play_left / cache_sbp); /* bytes needed until play_left == cache_left */
 
   // estimated playback time of current cached bytes
-  double cache_time = (static_cast<double>(cached) / currate) + (queueTime / 1000.0);
+  const double cacheTime = (static_cast<double>(cached) / currate) + (queueTime / 1000.0);
+
+  // cache level as current forward bytes / max forward bytes [0.0 - 1.0]
+  const double cacheLevel = (maxforward > 0) ? static_cast<double>(cached) / maxforward : 0.0;
```

Comment:
Shouldn't we also consider what's in our video-queue?

Reply:
I'm not sure because is different of playback time:

playback time = estimated playback time of cache + queue time 

but 

"cache level" only depends of cache

Anyway when cache is disabled Video Player uses queue time instead here:

https://github.com/xbmc/xbmc/blob/4afe993fe0ec92ed1d1c5f73bf1937a6ab6aa6dd/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L4926

Reply:
Then also "works" with cache disabled:

![osd](https://github.com/xbmc/xbmc/assets/58434170/fadfac38-6af1-428e-8ec1-c8ea7d1042b3)

Reply:
> I'm not sure because is different of playback time:
> 
> playback time = estimated playback time of cache + queue time
> 
> but
> 
> "cache level" only depends of cache
> 
> Anyway when cache is disabled Video Player uses queue time instead here:
> 
> https://github.com/xbmc/xbmc/blob/4afe993fe0ec92ed1d1c5f73bf1937a6ab6aa6dd/xbmc/cores/VideoPlayer/VideoPlayer.cpp#L4926

Ah yes, that's probably what I vaguely remembered and triggered my comment. I'd like to know @fritsch 's opinion about this.